alvr/server_core/src/connection.rs

```diff
@@ -650,9 +650,19 @@ fn connection_pipeline(
             0
         };
 
+    // Sync the session settings with the modified OpenVR settings
+    // before sending them to the client
+    let mut new_session_settings = session_manager_lock.session().session_settings.clone();
+    new_session_settings.video.encoder_config.use_10bit = enable_10_bits_encoding;
+    new_session_settings.video.encoder_config.use_full_range = use_full_range;
+    new_session_settings.video.encoder_config.enable_hdr = enable_hdr;
+    new_session_settings.video.encoder_config.encoding_gamma = encoding_gamma;
+    let mut client_session = session_manager_lock.session().clone();
+    client_session.session_settings = new_session_settings;
+    
     dbg_connection!("connection_pipeline: send streaming config");
     let stream_config_packet = alvr_packets::encode_stream_config(
-        session_manager_lock.session(),
+        &&client_session,
```

Comment:
Why are you making a copy of the session?

Reply:
In short, when I used `session_manager_lock.session_mut`, it modified all the settings. Which arguably could be a beneficial side-effect since you'd see what the headset's preferences are, but I figured in this case it would be better to have a client-adjusted session config.

The technical reason is that the client reads the encoding gamma from the session config, so here when the client preferences and limitations (supporting AV1/high profile, etc) are merged into the OpenVR config, we also have to make a client session config which reflects those changes, and more importantly, the encoding gamma that OpenVR will actually be encoding with.

Reply:
Ahh i understand. But we don't do it this way. we keep negotiated values separated. You should use `NegotiatedStreamingConfig`